ext/opentelemetry-ext-wsgi/src/opentelemetry/ext/wsgi/__init__.py

```diff
@@ -46,14 +47,43 @@ def _add_request_attributes(span, environ):
         span.set_attribute("component", "http")
         span.set_attribute("http.method", environ["REQUEST_METHOD"])
 
-        host = environ.get("HTTP_HOST") or environ["SERVER_NAME"]
+        host = environ.get("HTTP_HOST")
+        if not host:
+            host = environ["SERVER_NAME"]
+            if environ["wsgi.url_scheme"] == "https":
+                if environ.get("SERVER_PORT", "443") != "443":
+                    host += ":" + environ["SERVER_PORT"]
+            elif environ.get("SERVER_PORT", "80") != "80":
+                host += ":" + environ["SERVER_PORT"]
```

Comment:
Ah I didn't realize that `http.host` was nonstandard! Thanks for catching this!
Given that it's not standard, should we just remove this? It seems like a lot of overhead per request to compute the host string.

Reply:
I don't think the calculation for host is particularly expensive and it could probably even be optimized. I'd wait until the data formats are reworked (this is something that will most likely happen).

Reply:
I shortened & hopefully optimized it.